src/Liquid.Repository.Mongo/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -18,57 +16,69 @@ public static class IServiceCollectionExtensions
         /// <summary>
         /// Registers a <see cref="MongoRepository{TEntity, TIdentifier}"/> for the entity <typeparamref name="TEntity"/>,
         /// and a <see cref="MongoClientFactory"/>  if not previously registered.
-        /// This method also registers <see cref="LiquidTelemetryInterceptor"/> for MongoRepository instance.
+        /// This method may also registers <see cref="LiquidTelemetryInterceptor"/> for MongoRepository instance.
         /// </summary>
         /// <typeparam name="TEntity">Type of entity that the repository should correspond to</typeparam>
         /// <typeparam name="TIdentifier">Entity identifier type.</typeparam>
         /// <param name="services">Extended ServiceCollection object.</param>
-        /// <param name="entityOptions">MongoEntityOptions to configure how the entity will be persisted on Mongo.</param>
-        public static IServiceCollection AddLiquidMongoWithTelemetry<TEntity, TIdentifier>(this IServiceCollection services, Action<MongoEntityOptions> entityOptions)
+        /// <param name="entitiesConfigurationRootSectionName">Name of the configuration section where all entities have their repository settings configured. Default: "Liquid:RepositorySettings:Entities".</param>
+        /// <param name="activateTelemetry">Specifies whether the telemetry should be activated or not for this repository. Default: True.</param>
+        public static IServiceCollection AddLiquidMongoRepository<TEntity, TIdentifier>(this IServiceCollection services, string entitiesConfigurationRootSectionName = "Liquid:RepositorySettings:Entities", bool activateTelemetry = true)
             where TEntity : LiquidEntity<TIdentifier>, new()
         {
-            if (services.FirstOrDefault(x => x.ServiceType == typeof(IMongoClientFactory)) is null)
-                services.AddSingleton<IMongoClientFactory, MongoClientFactory>();
-
-            services.AddOptions();
-
-            services.Configure<MongoEntityOptions>(typeof(TEntity).Name, entityOptions);
+            services.TryAddSingleton<IMongoClientFactory, MongoClientFactory>();
+            services.TryAddSingleton<IMongoEntitySettingsFactory>(provider => { return ActivatorUtilities.CreateInstance<MongoEntitySettingsFactory>(provider, entitiesConfigurationRootSectionName); });
 
             services.AddScoped<IMongoDataContext<TEntity>, MongoDataContext<TEntity>>();
 
             services.AddScoped<MongoRepository<TEntity, TIdentifier>>();
 
-            services.AddLiquidTelemetryInterceptor<ILiquidRepository<TEntity, TIdentifier>, MongoRepository<TEntity, TIdentifier>>();
+            if (activateTelemetry)
+                services.AddLiquidTelemetryInterceptor<ILiquidRepository<TEntity, TIdentifier>, MongoRepository<TEntity, TIdentifier>>();
```

Comment:
This method has been deprecated in the latest stable version of Liquid.Core. Please update the package reference, and use the new method that is most suitable.